packages/Collapsible/src/Collapsible.js

```diff
@@ -52,7 +53,16 @@ const defaultProps = {
 const Collapsible = props => {
   const I18n = useI18n();
   let hasWarnedForA11yText = false;
-  const collapsedIcon = [props.iconExpand, props.iconCollapse];
+
+  const [collapsedIcon, setCollapsedIcon] = React.useState([props.iconExpand, props.iconCollapse]);
+
+  React.useEffect(() => {
+    if (props.iconAlign === "right") {
+      setCollapsedIcon([<UpArrowIcon />, <DownArrowIcon />]);
+    } else {
+      setCollapsedIcon([props.iconExpand, props.iconCollapse]);
+    }
+  }, []);
```

Comment:
I think `collapsedIcon` doesn't need to be a state, and using `useEffect` here. The default values of `iconExpand`, `iconCollapse` can be `null`, and then doing something like this:

```
const collapsedIcon = [props. iconExpand || props.iconAlign === "right" ? <UpArrowIcon /> : <RightArrowIcon /> , props. iconCollapse || props.iconAlign === "right" ? <DownArrowIcon /> : <RightArrowIcon />]
```